app/src/server/utils/prepareDatasetEntriesForImport.ts

```diff
@@ -26,12 +29,10 @@ export const prepareDatasetEntriesForImport = async (
 
   const trainingRatio = dataset?.trainingRatio ?? 0.8;
 
-  existingTrainingCount += entriesToImport.filter((row) => row.split === "TRAIN").length;
-
   const newTotalEntries = existingCount + entriesToImport.length;
   const numTrainingToAdd = Math.floor(trainingRatio * newTotalEntries) - existingTrainingCount;
 
-  let numTrainingAdded = 0;
+  let numTrainingAdded = entriesToImport.filter((row) => row.split === "TRAIN").length;
```

Comment:
I think it's more clear to set `numTrainingAdded` to actually encompass the total number of training rows added in this call. Same goes for `existingTrainingCount`, it should be equal to the number of training rows that existed before this route was called.

Reply:
I agree that with your new changes the code is cleaner than the way I wrote it initially.